index.bs

```diff
@@ -786,26 +805,60 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
 
 					1. Abort these steps.
 
-				1. Let |data| be a copy of the [=system clipboard data=] represented as [=clipboard items=]. For the MIME types defined in the [=mandatory data types=] list, |data| MAY be sanitized, but image/png format has unsanitized payload to preserve meta data.
+				1. Let |data| be a copy of the [=system clipboard data=].
 
-					Issue: Some OSs contain multiple clipboard (e.g. Linux, "primary", "secondary", "selection"). Define from which of those data is read.
+				1. Let |items| be a [=sequence=]&lt;[=/clipboard item=]&gt;.
+
+				1. For each |systemClipboardItem| in |data|:
+
+					1. Let |item| be a new [=/clipboard item=].
+
+					1. For each |systemClipboardRepresentation| in |systemClipboardItem|:
+
+						1. Let |mimeType| be the result of running the [=well-known mime type from os specific format=] algorithm given |systemClipboardRepresentation|'s [=system clipboard representation/name=].
+						
+						1. If |mimeType| is null, continue this loop.
+
+						1. Let |representation| be a new [=representation=].
+						
+						1. Set |representation|'s [=representation/MIME type=] to |mimeType|.
 
-					Issue: Add definition of sanitized.
+						1. Set |representation|'s [=representation/data=] to |systemClipboardRepresentation|'s [=system clipboard representation/data=].
+
+							Issue: It should be possible to read the data asynchronously from the system clipboard after the author calls getType, however, this set of steps implies that data will be provided at the time of read.
+						
+						1. The user agent, MAY sanitize |representation|'s [=representation/data=], unless |representation|'s [=representation/MIME type=]'s essence is "image/png", which should remain unsanitized to preserve meta data.
+
+						1. Append |representation| to |item|'s [=list of representations=].
+					
+					1. If |item|'s [=list of representations=] size is greater than 0, append |item| to |items|.
+					
+				If |items| has a size > 0, then:
+				
+					1. Let |firstItem| be |items|[0]
+					
+					1. Run the [=read web custom format=] algorithm given |firstItem|.
```

Comment:
Per https://w3c.github.io/clipboard-apis/#dom-clipboarditem-types, `type` getter of ClipboardItem returns `types array`, but it is not clear how does `types array` be constructed. It would be nice that we could have some steps for that.

Not sure if this makes sense: maybe we could construct `types array` from ` list of representations`, but which means we would need to read custom types first as [we would like the custom types listed at the top](https://github.com/w3c/clipboard-apis/issues/167#issuecomment-1031748387).

Reply:
I agree. We should definitely add this to the spec.